solr/solrj/src/java/org/apache/solr/common/cloud/Replica.java

```diff
@@ -70,16 +75,24 @@
      * cluster and it's state should be discarded.
      * </p>
      */
-    RECOVERY_FAILED;
-    
+    RECOVERY_FAILED("F");
+
+    /**short name for a state. Used to encode this in the state node see {@link PerReplicaStates.State}
+     */
+    public final String shortName;
+
+    State(String c) {
+      this.shortName = c;
+    }
+
     @Override
     public String toString() {
       return super.toString().toLowerCase(Locale.ROOT);
```

Comment:
What is the value of having upper case here if we translate to lower case elsewhere?

Reply:
The shortname is not the same as name.